### DISCUSSION TITLE
Why SWR is not refetching data after remounting component?

### DISCUSSION THREAD

**Discussion by skorphil** [vercel/swr#2842] [Q&A]
Hi, i'm struggling to understand how SWR works after all. Let's start with overview what I have:
**Parent.jsx** - component with button. When I press button, the `<Children />` component is mounted
```jsx
export function Parent() {
  const [mountCounter, setMountCounter] = useState(0);
  const [mountedChild, setMountedChild] = useState(false);

const closeTest = () => {
    setMountedChild(false);
  };

  const openTest = () => {
    setMountedChild(true);
  };

  let childComponent = mountedChild ? <Child unmountSelf={closeTest} /> : "";
  return (
    <>
      <button onClick={openTest}>Open Child</button>
      <div>---</div>
      {childComponent}
    </>
  );
}
```

**Child.jsx** - component with has to fetch data from API. Also it can be closed with a button, which use method from `Parent.jsx`
```jsx
  export function Child({ unmountSelf }) {
  const [mountCounter, setMountCounter] = useState(0);
  
  let { data, error, isLoading } = useSWR(
    `https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/latest/currencies.json`,
    fetcher,
    {
      revalidateOnFocus: false,
      revalidateIfStale: true,
      revalidateOnMount: true,
      onSuccess: (data) => {
      },
    },
  );

  if (isLoading) {
    return <div>Loading</div>;
  }

  let contextValue = data.usd;

  return (
    <>
      <TestContext.Provider value={contextValue}>
        <div>{contextValue}</div>
        <button onClick={unmountSelf}>Close Child</button>
      </TestContext.Provider>
    </>
  );
}

const fetcher = (...args) => fetch(...args).then((res) => res.json());
```

Well, when I first click button, so `Child` being mounted for the first time, everything is working as expected:
1. SWR is fetching data
2. SWR isLoading == true
3. After success the onSuccess code is executed
...
- Finally, Child component get to return statement and returned fetched content

![Screenshot 2023-11-26 at 21 02 20](https://github.com/vercel/swr/assets/6762009/4b47d01f-c646-4489-8d0d-c675036dcffb)

But, if I close (press button) and open child component again, the behavior becomes very illogical:
1. The component immediately get to return statement without fetching
2. useSwr never returns isLoading state
...
- At the end onSuccess SWR code is executed

![Screenshot 2023-11-26 at 21 06 53](https://github.com/vercel/swr/assets/6762009/95a22575-c1b5-4526-8179-5534e0c29306)

You can play with this stuff here - [Codesandbox](https://codesandbox.io/p/devbox/angry-surf-yn4z9n?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpfp28l300093b6f4seyv8lq%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpfp28l300033b6fekkx81t1%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpfp28l300063b6fud3qbd83%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpfp28l300083b6fnvxyfqoa%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B64.87274596116461%252C35.12725403883539%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpfp28l300033b6fekkx81t1%2522%253A%257B%2522id%2522%253A%2522clpfp28l300033b6fekkx81t1%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpfp28l300083b6fnvxyfqoa%2522%253A%257B%2522id%2522%253A%2522clpfp28l300083b6fnvxyfqoa%2522%252C%2522activeTabId%2522%253A%2522clpfppaot01pr3b6f9rusj6hf%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clpfppaot01pr3b6f9rusj6hf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clpfp28l300063b6fud3qbd83%2522%253A%257B%2522id%2522%253A%2522clpfp28l300063b6fud3qbd83%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clpfpp6yd01o53b6fzqvrubqe%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clpfpp6yd01o53b6fzqvrubqe%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) (Console output worked for me only if open preview in separated tab)

Can someone explain me why swr not refetching normally?

**Comment by sshellmann** [comment]
Late response and I'm just making a guess since I'm not an expert with SWR but I would assume that it's just returning the cached value. If you need it to get a new value each time, you can look at someone with a similar problem here:

https://github.com/vercel/swr/discussions/456#discussioncomment-1006011
